### PR TITLE
Trying to backport #18228 to 4-2-stable

### PR DESCRIPTION
**Pull request by joallard**
I couldn't get a Travis run on my repo, so here. #18228

ie Is it possible to backport 18228 to 4-2?

### CODE REVIEW

File: activerecord/lib/active_record/connection_adapters/abstract_mysql_adapter.rb [hunk 1]
```diff
@@ -58,6 +58,18 @@ def schema_creation
...
       end
 
+      def column_spec_for_primary_key(column)
+        spec = {}
+        if column.extra == 'auto_increment'
+          return unless column.limit == 8
+          spec[:id] = ':bigint'
+        else
+          spec[:id] = column.type.inspect
+          spec.merge!(prepare_column_options(column).delete_if { |key, _| [:name, :type, :null].include?(key) })
```
Comment (Envek) [inline activerecord/lib/active_record/connection_adapters/abstract_mysql_adapter.rb]: Code fails there as `prepare_column_options` in 4.2 requires two arguments: `column` and `types`. Second argument must be supplied (I do not know what is it).

ActiveRecord internals differs heavily between 4.2 and 5.0, so it wouldn't be easy to backport anything. But I need this fix too.

File: activerecord/lib/active_record/connection_adapters/postgresql_adapter.rb [hunk 2]
```diff
@@ -125,6 +125,21 @@ def schema_creation # :nodoc:
...
+        spec = {}
+        if column.serial?
+          return unless column.sql_type == 'bigint'
+          spec[:id] = ':bigserial'
+        elsif column.type == :uuid
+          spec[:id] = ':uuid'
+          spec[:default] = column.default_function.inspect
+        else
+          spec[:id] = column.type.inspect
+          spec.merge!(prepare_column_options(column).delete_if { |key, _| [:name, :type, :null].include?(key) })
```
Comment (Envek) [inline activerecord/lib/active_record/connection_adapters/postgresql_adapter.rb]: Same thing here: `prepare_column_options` in 4.2 requires two arguments: `column` and `types`.

**Comment by joallard** [comment]
Closing in favor of #21940, looks like @carols10cents has done great work!

**Comment by carols10cents** [comment]
@joallard thanks! you gave me a great start :)

**Pull request closed without merging**